### ISSUE TITLE
improve resolver errors in trivial error cases

### ISSUE THREAD

**Issue by rmadsen-ks** [opentap/opentap#1525]
## Multiple packages with the same name but different versions specified.

This can happen in build scenarios where for example a OpenTapPackageReference is made in the Directory.Build.props file and the csproj file. The error message can be confusing:
```
 error OpenTAP Install: Unable to resolve image 'Sign:1.4.0, Sign:1.5.0-alpha.4.9+6a5355a8.sign-apple-apps, 
Dotfuscator:2.0.2-beta.6+11a5bc9a, OpenTAP:9.23.2+5a906a86': 
Sign: 1.4.0, Sign: 1.5.0-alpha.4.9+6a5355a8.sign-apple-apps was not found
```

Actually it was found, but the constraint cannot be solved. because both Sign package references are required to be that version.

## The package names does not exist

```
PS C:\git\opentap> ~/tap/tap image install --dry-run abc:1.2.3
Unable to resolve image 'abc:1.2.3': abc: 1.2.3 was not found
```

In this case, the problem is not the version, but no package exists with that name. It would be better to point that out.

**Comment by sebastian-pop** [comment]
@Oeh1995 could I get a test scenario for the first issue?

for the second issue the message is changed but it still implies a problem with the version not that the package does not exist. Perhaps something like `abc was not found` instead of `abc: 1.2.3 was not found`



![Image](https://github.com/opentap/opentap/assets/97734470/7a417b73-2aea-48a7-860e-de33cdd11f97)

Tested on OpenTAP 9.25.0-beta.47

**Comment by Oeh1995** [comment]
@alnlarsen could you elaborate on this?

**Comment by sebastian-pop** [comment]
Reopening.

- First issue: fix does not work

Command: `tap image install "Sign:1.4.0, Sign:1.5.0-alpha.4.9+6a5355a8.sign-apple-apps, OpenTAP:9.23.2+5a906a86, Dotfuscator:2.0.2-beta.6+11a5bc9a" -r "https://test-automation.pw.keysight.com/api/packages;token=ADD_TOKEN_HERE`


![Image](https://github.com/opentap/opentap/assets/97734470/79eec672-f6b4-4353-bc4b-43f7736fa9a3)

- second issue. improve the error message to imply the package was not found

**Comment by sebastian-pop** [comment]
Fixed on 9.25.0-rc.1

1st scenario: 

![Image](https://github.com/opentap/opentap/assets/97734470/4009de40-b93d-4082-be2e-ce1f28813bfb)

2nd scenario

![Image](https://github.com/opentap/opentap/assets/97734470/366ed13c-1a12-4f1b-a3fe-877eaeda8165)
